draft: create more constructors to use base Entity constructors

maeg02 opened the pull request:
DRAFT ONLY, to start the discussion of how to solve #204

Could there be an AppSetting to control this?
Now it's only if(true)

Is there a smarter way to know if the constructors are already there than based on Length <= 1?

maeg02 commented:
Yes, however it was way past my bedtime and I still hacked away on it, so quality is guaranteed, I will make a manual test, to see if the overloads work as intended with AltKey. You know, more than *it compiles* let shit this... (as is the current state)

daryllabar commented on DLaB.CrmSvcUtilExtensions/Entity/EntityConstructorsGenerator.cs:
```diff
@@ -0,0 +1,64 @@
+        public void CustomizeCodeDom(CodeCompileUnit codeUnit, IServiceProvider services)
+        {
+            var metadataService = (IMetadataProviderService)services.GetService(typeof(IMetadataProviderService));
+            var metadata = metadataService.LoadMetadata();
```
Looks like this isn't used?

maeg02 replied:
Ahm... true. Sharp eyes!
Should I remove it?

daryllabar replied:
Yes Please

daryllabar commented on DLaB.CrmSvcUtilExtensions/Entity/CustomizeCodeDomService.cs:
```diff
@@ -32,6 +32,10 @@ public void CustomizeCodeDom(CodeCompileUnit codeUnit, IServiceProvider services
             {
                 new CodeCustomization(Parameters).CustomizeCodeDom(codeUnit, services);
             }
+            if(true)
```
So the way this works, is you set a flag in the config that its read as a property.  Then I do a search for the app.Config key of a similar value, and find where it all is updated, and create the new property in the UI/Settings of the actual EBG

maeg02 replied:
yeah, sounds like a plan, if you want to do that, that would be awesome!
Naming things is hard, so if you have something in mind, then that would be good.
Would it be good if the new config would default to false?

daryllabar replied:
I can do it.  When is just the question...

As far as naming things, GenerateConstructorsSansLogicalName?

maeg02 commented:
An update with the new config, ( a bunch of files now changed)
And update with the actual code:
Moved it earlier in the generated class, and removed unused code
